test/foundry/local/Blast_DaiRetriever.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Test } from "forge-std/Test.sol";
+import { MockERC20 } from "forge-std/mocks/MockERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+import "forge-std/console.sol";
+
+import { Blast_DaiRetriever } from "../../../contracts/Blast_DaiRetriever.sol";
+import { MockBlastUsdYieldManager } from "../../../contracts/test/MockBlastUsdYieldManager.sol";
+
+contract Token_ERC20 is ERC20 {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
+
+    function mint(address to, uint256 value) public virtual {
+        _mint(to, value);
+    }
+}
+
+contract BlastDaiRetrieverTest is Test {
+    Blast_DaiRetriever daiRetriever;
+    MockBlastUsdYieldManager usdYieldManager;
+    Token_ERC20 dai;
+
+    // EOA caller of retrieve().
+    address rando;
+
+    // HubPool should receive funds.
+    address hubPool;
+
+    uint256 requestId = 2;
+    uint256 hintId = 3;
+
+    function setUp() public {
+        dai = new Token_ERC20("DAI", "DAI");
+        rando = vm.addr(1);
+        hubPool = vm.addr(2);
+
+        usdYieldManager = new MockBlastUsdYieldManager();
+        daiRetriever = new Blast_DaiRetriever(hubPool, usdYieldManager, IERC20Upgradeable(address(dai)));
+    }
+
+    function testRetrieveSuccess() public {
+        assertEq(dai.balanceOf(hubPool), 0);
+        assertEq(dai.balanceOf(rando), 0);
+        uint256 daiTransferAmount = 10**22;
+        dai.mint(address(hubPool), daiTransferAmount);
+        vm.startPrank(rando);
+
+        // Make sure claimWithdrawal is called with correct params and returns true.
+        vm.expectCall(address(usdYieldManager), abi.encodeCall(usdYieldManager.claimWithdrawal, (requestId, hintId)));
+        vm.expectEmit(true, true, false, true, address(usdYieldManager));
+        emit MockBlastUsdYieldManager.ClaimedWithdrawal(requestId, hintId);
```

Comment:
I'm not sure what this is testing.

Reply:
it tests that the function is called, but actually since we use the `expectCall` its redundant so I'll remove